stablehlo/tests/interpret_sign.mlir

```diff
@@ -0,0 +1,168 @@
+// RUN: stablehlo-translate --interpret -split-input-file %s
+
+// CHECK-LABEL: Evaluated results of function: sign_op_test_si4
```

Comment:
Per this [checklist](https://github.com/openxla/stablehlo/blob/main/docs/reference.md?plain=1#L155-L157), we only need to test `si64`, `f64`, and `c128`.